### PR TITLE
Clear preferences test rule

### PR DESCRIPTION
**Pull request by Sloy**
This rule lets us clear all preferences from the app between each test method. This way they are independent of any side-effects happening in any test.

TODO: 
- [x] Add Javadoc to the rule
- [x] Explain in the readme

### CODE REVIEW

File: sample/src/androidTest/java/com/schibsted/spain/barista/sample/ClearPreferencesRuleTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+public class ClearPreferencesRuleTest {
+
+  @Rule
+  public ActivityTestRule<PreferencesActivity> activityRule = new ActivityTestRule<>(PreferencesActivity.class, true, false);
+
+  @Rule
+  public ClearPreferencesRule clearPreferencesRule = new ClearPreferencesRule();
+
+  //
+  // Only one of these two tests will succeed when preferences are not cleared.
```
Comment (rocboronat) [inline sample/src/androidTest/java/com/schibsted/spain/barista/sample/ClearPreferencesRuleTest.java]: Great comment there :·)

File: sample/src/androidTest/java/com/schibsted/spain/barista/sample/ClearPreferencesRuleTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  // They rely on preferences being cleared between test executions.
+  //
+
+  @Test
+  public void checkOnceThatValueIsZeroFistAndOneAfterIncrement() throws Exception {
+    activityRule.launchActivity(null);
+
+    assertCurrentValueIs("0");
+    incrementValue();
+    assertCurrentValueIs("1");
```
Comment (rocboronat) [inline sample/src/androidTest/java/com/schibsted/spain/barista/sample/ClearPreferencesRuleTest.java]: You're a poet! <3

**Comment by rocboronat** [comment]
Great work there @Sloy ! Including that smart way to test the @Rule :·)

**Comment by rocboronat** [comment]
@Sloy thanks! One more check?

**Comment by Sloy** [comment]
Awesome! Thanks so much for documenting it @rocboronat, I know it's a boring task.
👍

**Comment by rocboronat** [comment]
💋

**Pull request merged**